aws/rust-runtime/aws-runtime/src/retries/classifier.rs

```diff
@@ -32,15 +35,45 @@ pub const TRANSIENT_ERRORS: &[&str] = &["RequestTimeout", "RequestTimeoutExcepti
 
 /// A retry classifier for determining if the response sent by an AWS service requires a retry.
 #[derive(Debug)]
-pub struct AwsErrorCodeClassifier;
-
-impl AwsErrorCodeClassifier {
-    /// Classify an error code to check if represents a retryable error. The codes of retryable
-    /// errors are defined [here](THROTTLING_ERRORS) and [here](TRANSIENT_ERRORS).
-    pub fn classify_error<E: ProvideErrorMetadata, R>(
-        &self,
-        error: &SdkError<E, R>,
-    ) -> Option<RetryReason> {
+pub struct AwsErrorCodeClassifier<E>
+where
+    E: StdError + ProvideErrorMetadata + Send + Sync + 'static,
```

Comment:
What do you think about just passing in the error metadata rather than making it generic over `E`?

Reply:
The retry strategy is responsible for calling the classifiers and it does so by calling a single wrapper struct that works similar to how the `Interceptors` wrapper thing worked. The classifiers are all stored within this struct, which itself is stored in the config bag. So, either we can use an API where everything has the same args, or we can codegen the "access the right type for a classifier and then pass it to the classifier" stuff.

I wanted to avoid generating more code since this would affect all operations. Hence, I prefer the current approach.